Dockerfile

```diff
@@ -41,7 +41,15 @@ RUN : \
  && create-db.sh aplus aplus django-migrate.sh \
  && :
 
+RUN pip_install debugpy git+https://github.com/mher/flower#egg=flower
```

Comment:
Each `RUN` instruction creates a new layer in the built container image. The old RUN instruction (that installs the A-plus source code above) has avoided adding many layers by including "too many" commands in a single RUN. Did you consider what would be the optimal layout for the layers?

One principle is that the "more common" parts that change less often are placed in the bottom layers. If we assume that the A-plus source code changes more often than the `debugpy` pip package in our use case, then `RUN pip_install debugpy ...` could be placed before the `RUN` instruction for installing A-plus.

Creating the directory `var/celery/results` is basically always the same so it could be in the top of the Dockerfile in the lower image layers.

https://docs.docker.com/engine/reference/builder/#run

Reply:
I moved debugpy install earlier, but could not move /var/celery/results because it depends on aplus user that is created in the large RUN block. We could combine it to the large block, though.